### PR TITLE
Bootstrap hooks and DebugKit Integration

### PR DESCRIPTION
**Pull request by markstory**
Add bootstrap hooks and DebugKit compatibility. I've also made this plugin work with PaginatorComponent once the pull request to update `RepositoryInterface` is merged.

After this, I think we can tag a dev preview release as the plugin is usable in simple CRUD scenarios.

### CODE REVIEW

File: config/bootstrap.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    $view = $event->subject();
+    $view->Form->addContextProvider('elastic', function ($request, $data) {
+        $first = null;
+        if (is_array($data['entity']) || $data['entity'] instanceof Traversable) {
+            $first = (new Collection($data['entity']))->first();
+        }
+        if ($data['entity'] instanceof Document || $first instanceof Document) {
+            return new DocumentContext($request, $data);
+        }
+    });
```
Comment (lorenzo) [inline config/bootstrap.php]: Clever!

File: src/Datasource/Connection.php [hunk 2]
```diff
@@ -15,7 +15,7 @@
 namespace Cake\ElasticSearch\Datasource;
 
 use Cake\ElasticSearch\Datasource\SchemaCollection;
-use Cake\Log\Log;
+use Cake\Database\Log\LoggedQuery;
```
Comment (lorenzo) [inline src/Datasource/Connection.php]: I think we should move this class to `Datasource`

File: config/bootstrap.php (hunk 1 above)
Comment (lorenzo) [inline config/bootstrap.php]: Is this also triggered for each element?

File: src/Datasource/Connection.php (hunk 2 above)
Comment (markstory) [inline src/Datasource/Connection.php]: Yes, and have an interface for it. Right now the logger hints against the implementation which is kind of annoying.

File: config/bootstrap.php (hunk 1 above)
Comment (markstory) [inline config/bootstrap.php]: No, but before each cell. Which is unfortunate.

**Comment by lorenzo** [comment]
Great job @markstory !

**Pull request merged**